services/hexathons/src/routes/visit.ts

```diff
@@ -0,0 +1,138 @@
+/* eslint-disable no-underscore-dangle */
+import { apiCall, asyncHandler, BadRequestError, checkAbility } from "@api/common";
+import { Service } from "@api/config";
+import express from "express";
+import { FilterQuery } from "mongoose";
+
+import { VisitModel, Visit } from "../models/visit";
+
+export const visitRouter = express.Router();
+
+// get visit
+visitRouter.route("/").get(
+  checkAbility("read", "Visit"),
+  asyncHandler(async (req, res) => {
+    const company = await apiCall(
+      Service.USERS,
+      { method: "GET", url: `/companies/employees/${req.user?.uid}` },
+      req
+    );
+    const filter: FilterQuery<Visit> = {};
+    if (req.query.hexathon) {
+      filter.hexathon = String(req.query.hexathon);
+    }
+    if (req.query.visitorId) {
+      filter.visitorId = String(req.query.visitorId);
+    }
+    if (company) {
+      filter.company = String(company.id);
+    }
+
+    const visits = await VisitModel.accessibleBy(req.ability).find(filter);
+    return res.status(200).send(visits);
+  })
+);
+
+visitRouter.route("/:visitorId").get(
+  checkAbility("read", "Visit"),
+  asyncHandler(async (req, res) => {
+    // get company of current user
+    const company = await apiCall(
+      Service.USERS,
+      { method: "GET", url: `/companies/employees/${req.user?.uid}` },
+      req
+    );
+
+    if (company) {
+      const { visitorId } = req.params;
+      const visit = await VisitModel.findOne({
+        visitorId,
+        company: company.id,
+      });
+
+      return res.status(200).send(visit?.toJSON());
+    } 
+      throw new BadRequestError("Current user not associated with a company");
+    
+  })
+);
+
+visitRouter.route("/:visitorId").post(
```

Comment:
in general, it's bad practice to query params in a post request. So don't use `:visitorId` here, just move it to the POST body, so like `req.body.visitorId`.